src/core/border_router/routing_manager.cpp

```diff
@@ -603,47 +684,75 @@ void RoutingManager::EvaluateNat64Prefix(void)
 {
     OT_ASSERT(mIsRunning);
 
-    NetworkData::Iterator            iterator = NetworkData::kIteratorInit;
-    NetworkData::ExternalRouteConfig config;
-    Ip6::Prefix                      smallestNat64Prefix;
+    Error                            error;
+    NetworkData::ExternalRouteConfig preferredNat64PrefixConfig;
 
     LogInfo("Evaluating NAT64 prefix");
 
-    smallestNat64Prefix.Clear();
-    while (Get<NetworkData::Leader>().GetNextExternalRoute(iterator, config) == kErrorNone)
+    error = Get<NetworkData::Leader>().GetPreferredNat64Prefix(preferredNat64PrefixConfig);
+
+    // (Re)advertise NAT64 prefix from this BR when
+    // - no NAT64 prefix exits in Network Data yet
+    // - the preferred NAT64 prefix in Network Data was advertised by this BR
+    if (error == kErrorNotFound || preferredNat64PrefixConfig.GetPrefix() == mAdvertisedNat64Prefix)
```

Comment:
```suggestion
    // - the preferred NAT64 prefix in Network Data was advertised by this BR
    if (error == kErrorNotFound || preferredNat64PrefixConfig.mRloc16() == <Local RLOC16>)
```

I think it's better to check RLOC16 to determine if the NAT64 prefix was advertised by the BR itself. I think it's possible that preferredNat64PrefixConfig.GetPrefix() is different from mAdvertisedNat64Prefix.